nixos/doc/manual/release-notes/rl-2505.section.md

```diff
@@ -70,6 +70,8 @@
 
 - `nodePackages.ganache` has been removed, as the package has been deprecated by upstream.
 
+- `services.waagent` option has been added, making `/etc/waagent.conf` customizable. `virtualisation.azure.agent` option provided by `azure-agent.nix` module has been deprecated and the module will be removed in a future release.
```

Comment:
I think this first part should be split into the "New Modules" section, mentioning the latter here